src/main/java/nb/kafka/operator/TopicManager.java

```diff
@@ -47,13 +49,10 @@ public NewTopic createTopic(Topic topic) throws TopicCreationException {
   }
 
   public void updateTopic(Topic newTopic) throws InterruptedException, ExecutionException {
-    PartitionnedTopic oldTopic = describeTopic(newTopic.getName());
-    if (newTopic.getPartitions() < oldTopic.getPartitions()) {
-      throw new IllegalArgumentException("Reduction of the number of partitions is not allowed");
-    }
-    if (newTopic.getReplicationFactor() != 0 && newTopic.getReplicationFactor() != oldTopic.getReplicationFactor()) {
-      throw new IllegalArgumentException("Change of the replication factor is not supported");
-    }
+    PartitionedTopic oldTopic = describeTopic(newTopic.getName());
+
+    if(!new TopicValidator(config, newTopic, oldTopic).isValid())
+      return;
```

Comment:
I think you overwritten a change here, it should be:
```
if (!new TopicValidator(config, newTopic, oldTopic).isValid()) {
  return;
}
```

Reply:
fixed in [1a56ae5](https://github.com/nbogojevic/kafka-operator/pull/24/commits/1a56ae5a7769f4338418127206f943a3a58c39fa)